mergify_engine/utils.py

```diff
@@ -185,23 +185,30 @@ def __init__(self, key: str, value: typing.Any):
         setattr(self, key, value)
 
 
-async def send_refresh(
+class PartialGitHubEventRefresh(typing.TypedDict, total=False):
+    pull_request_number: github_types.GitHubPullRequestNumber
+    ref: github_types.GitHubRefType
+    action: github_types.GitHubEventRefreshActionType
+
+
+async def _send_refresh(
     redis_cache: RedisCache,
     redis_stream: RedisStream,
     repository: github_types.GitHubRepository,
-    pull_request_number: typing.Optional[github_types.GitHubPullRequestNumber] = None,
-    ref: typing.Optional[github_types.GitHubRefType] = None,
-    action: github_types.GitHubEventRefreshActionType = "user",
+    refresh_options: PartialGitHubEventRefresh,
 ) -> None:
     # Break circular import
     from mergify_engine import github_events
 
+    if not refresh_options:
+        raise ValueError("refresh_options must be set")
```

Comment:
I'd rather remove the `total=False` and just pass `None` as value.
Or just go ahead and use keyword arguments, no?

Reply:
Hum, since I have one method of each kind of refresh I can keep keyword arguments I guess.